components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -195,26 +201,27 @@ func (sb *sensorBase) Reconfigure(ctx context.Context, deps resource.Dependencie
 
 	if sb.velocities != nil && len(newConf.ControlParameters) != 0 {
 		// assign linear and angular PID correctly based on the given type
-		var linear, angular control.PIDConfig
 		for _, c := range newConf.ControlParameters {
 			switch c.Type {
 			case typeLinVel:
-				linear = c
+				// configPIDVals at index 0 is linear
+				sb.configPIDVals[0] = c
 			case typeAngVel:
-				angular = c
+				// configPIDVals at index 1 is angular
+				sb.configPIDVals[1] = c
 			default:
-				sb.logger.Warn("control_parameters type must be 'linear_velocity' or 'angular_velocity'")
+				sb.logger.Error("control_parameters type must be 'linear_velocity' or 'angular_velocity'")
 			}
 		}
 
 		// unlock the mutex before setting up the control loop so that the motors
 		// are not locked, and can run if any auto-tuning is necessary
 		sb.mu.Unlock()
-		if err := sb.setupControlLoop(linear, angular); err != nil {
+		if err := sb.setupControlLoop(sb.configPIDVals[0], sb.configPIDVals[1]); err != nil {
```

Comment:
This may be better off as a validation check; if the control parameters for the sensor-controlled base are populated, let's check that they are of either type, and error if not so. 
We should allow the user to set one or the other, none or both, but error if neither match the two expected loop types.

Reply:
Is there a test (manual or software) that shows you what the experience looks like for a user right now, if they try to configure with 1) one badly named "type" 2) two badly named types? Are we hitting the tuning loops, and what happens if one loop is tuned but not the other in your current logic? What errors are returned?

Reply:
for the case that there's incorrect types, maybe we can return the error that we are currently only logging so that the component doesn't get built. otherwise it will probably be all 0's and try and autotune. 

if one loop is tuned and no the other, the same errors will be returned that is like "the base has been tuned copy these pid values" but it will only return the tuned one. for example if a user has linear values but the angular values are all 0 and angular is tuned, only the tuned angular pid values will be returned. same for do command.

Reply:
^ If the config is not nil, and there are incorrect types, and I copy over my PID values to those incorrect types, that's not great for a user, I'm assuming it'll just keep re-tuning even if I populate the fields of the malformed types?

Let's error in Validate for that case, tell people they've misspelt the types if the control part of the config is populated.

Reply:
Also, for Validation, add a test for all the Valid and erroring cases.

Reply:
I'm not fully understanding the issue. I can add a validation check still, but on line 213 we return an error that the type is invalid as well as providing the two valid types, and then the resource won't build. so you should hit this error if you have a typo even. I will add a test case for that to cover our bases and I can also add it to validation but we still need to go through this check here to assign the assign the linear and angular values correctly, so it kind of feels like repeated code unless I am misunderstanding the issue.

Reply:
Is it still the case that the builder will not let you complete the configuration on a component if there is a validation error? In that case that's a clear signal to the user that something is wrong, and that they have to fix the config first to be able to copy over values to tune. 

If this is not the current behaviour of this component's config builder card, then that's something to think about. If the resource graph cannot validate the resource then it will not try to build it in a loop, which it does with the constructor and reconfigure functions.

Reply:
okay added to validate as well